src/main.c

```diff
@@ -142,10 +147,160 @@ static bool is_valid_stream(char *s)
 	return true;
 }
 
+char* get_saved_file_path(void)
+{
+	const char* xdg_state_home = getenv("XDG_STATE_HOME");
+	const char* home_path = getenv("HOME");
+
+	if (!home_path & !xdg_state_home)
+	exit(EXIT_FAILURE);
+
+	const char* local_state = "/.local/state";
+	const char* dir_name = "/nudoku/";
+
+	char* dir_path = NULL;
+
+	if (xdg_state_home)
```

Comment:
This whole if block is duplication and will be improved by the above mentioned change :)